transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionRepository.java

```diff
@@ -0,0 +1,195 @@
+import org.pcollections.TreePVector;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static com.example.core.CompletionStageUtils.accept;
+import static com.example.core.CompletionStageUtils.doAll;
+
+@Singleton
+public class TransactionRepository {
+
+    private final CassandraSession session;
+
+    @Inject
+    public TransactionRepository(CassandraSession session, ReadSide readSide) {
+        this.session = session;
+        readSide.register(TransactionEventProcessor.class);
+    }
+
+    CompletionStage<PaginatedSequence<TransactionSummary>> getTransactionsForUser(
+            UUID userId, TransactionInfoStatus status, int page, int pageSize) {
+        return countTransactionSummary(userId, status)
+                .thenCompose(
+                        count -> {
+                            int offset = page * pageSize;
+                            int limit = (page + 1) * pageSize;
+                            CompletionStage<PSequence<TransactionSummary>> transactions = offset > count ?
+                                    CompletableFuture.completedFuture(TreePVector.empty()) :
+                                    selectTransactionSummary(userId, status, offset, limit);
+                            return transactions.thenApply(seq -> new PaginatedSequence<>(seq, page, pageSize, count));
+                        }
+                );
+    }
+
+    private CompletionStage<Integer> countTransactionSummary(UUID userId, TransactionInfoStatus status) {
+        return session
+                .selectOne(
+                        "SELECT COUNT(*) FROM transactionSummary " +
+                                "WHERE (creatorId = ? OR winnerId = ?) AND status = ? " +
+                                "ORDER BY itemId DESC",
+                        userId,
+                        userId,
+                        status
+                )
+                .thenApply(row -> (int) row.get().getLong("count"));
+    }
+
+    private CompletionStage<PSequence<TransactionSummary>> selectTransactionSummary(
+            UUID userId, TransactionInfoStatus status, long offset, int limit) {
+        return session
+                .selectAll(
+                        "SELECT * FROM transactionSummary " +
+                                "WHERE (creatorId = ? OR winnerId = ?) AND status = ? " +
+                                "ORDER BY itemId DESC " +
+                                "LIMIT ?",
+                        userId,
+                        userId,
+                        status,
+                        limit
+                )
+                .thenApply(List::stream)
+                .thenApply(rows -> rows.skip(offset))
+                .thenApply(rows -> rows.map(TransactionRepository::toTransactionSummary))
```

Comment:
Sorry for the delay. 
I think it's weird that a private method on a non-publicly facing class uses the public API. I'd find it awkward that changing the public API (imagine I renamed `TransactionSummary`) caused a change in `TransactionRepository`.

We can fix this later.